### ISSUE TITLE
Support for the HM10 Bluetooth Module

### ISSUE THREAD

**Issue by holger-dunst** [esphome/feature-requests#50]
The esp8266 can't do ble and the ble support of esp32 ist very unstable.
There is the cheap HM10 Bluetooth Module that can be used instead.
Have a look at [open mqtt gateway](https://github.com/1technophile/OpenMQTTGateway/wiki/Arduino-BT-Module) that supports it.
In addition it would be nice to have support for the [Xiaomi mi flora](https://1technophile.blogspot.com/2017/11/mi-flora-integration-to-openmqttgateway.html) sensors and the [Xiaomi Mijia Temperature and Humidity sensor](https://1technophile.blogspot.com/2018/05/openmqttgateway-now-reads-xiaomi-mijia.html).

**Comment by blademckain** [comment]
if esphome could support the HM10 module it would be great!

**Comment by james-fry** [comment]
> Xiaomi MiJia will be supported in the next release

Hi @OttoWinter 
Please can you advise (if you know): Are both versions of the xiaomi BLE temp device supported?

![image](https://user-images.githubusercontent.com/5823294/56806962-59222780-6825-11e9-9df2-02269021c4b7.png)

In the docs it only shows the LCD version, but the "cleargrass" version with epaper looks v nice.

**Comment by OttoWinter** [comment]
> Please can you advise (if you know): Are both versions of the xiaomi BLE temp device supported?

Please don't use this feature request for that question, this is strictly about the HM-10. Anyway, I don't if it is supported or not - it looks similar so chances are it works, but that is not certain.

**Comment by james-fry** [comment]
My bad to use this to ask - i guess I didn’t read it properly and saw the mijia devices mentioned. Thx for the info

**Comment by itProfi** [comment]
Dear friends! Is today esphome support HM-10 or no?

**Comment by OttoWinter** [comment]
@itProfi No, it's not - and because the ESP32 BLE implementation has gotten much more stable over time it's also getting less likely.

Also, I don't understand what you mean by those devices. Do you want thermostats based on the ESP8266 with Bluetooth support to run ESPHome? Why do you need bluetooth then, when you already have wifi? The main reason for BLE support was (and is) integrating third party devices that operate with bluetooth into your existing home system.

**Comment by Jpsy** [comment]
Well an example of such a "ready-made" device are Tuya smart power plugs that are based on ESP8266. They are ideal for adding additional logic right within their casing. With an HM10 they could become a BLE-2-Wifi bridge that you can simply plug in anywhere near your BLE devices.
